pittapi/sports.py

```diff
@@ -56,37 +36,56 @@ def get_mens_basketball_record() -> str:
     return record_summary
 
 
-def get_next_mens_basketball_game() -> NextBasketballGame:
+def get_next_mens_basketball_game() -> dict:
     """returns a dict containing details of the next scheduled men's basketball game."""
-    basketball_response = requests.get(BASKETBALL_URL)
-    basketball_data = basketball_response.json()
-
-    next_game = NextBasketballGame(
-        name=basketball_data["team"]["nextEvent"][0]["name"],
-        short_name=basketball_data["team"]["nextEvent"][0]["shortName"],
-        season_name=basketball_data["team"]["nextEvent"][0]["seasonType"]["name"],
-        week=basketball_data["team"]["nextEvent"][0]["week"]["text"],
-        court=basketball_data["team"]["nextEvent"][0]["competitions"][0]["venue"][
-            "fullName"
-        ],
-    )
-
-    return next_game
-
-
-def get_mens_basketball_standings():
+    basketball_data = _get_mens_basketball_data()
+    next_game = None
+    try:
+        next_game = basketball_data["team"]["nextEvent"][0]
+        opponent = None
+        homeaway = None
+        status = None
+        if next_game["competitions"][0]["status"]["type"]["name"] == "STATUS_FINAL":
+            status = "GAME_COMPLETE"
+        elif next_game["competitions"][0]["status"]["type"]["name"] == "STATUS_IN_PROGRESS":
+            status = "IN_PROGRESS"
+        if next_game["competitions"][0]["competitors"][0]["id"] == 221:
+            opponent = next_game["competitions"][0]["competitors"][0]
+            homeaway = next_game["competitions"][0]["competitors"][0]["homeAway"]
+        else:
+            opponent = next_game["competitions"][0]["competitors"][1]
+            homeaway = next_game["competitions"][0]["competitors"][1]["homeAway"]
+        return {
+            "timestamp" : next_game["date"],
+            "opponent" : {
+                "id" : opponent["team"]["id"],
+                "school" : opponent["team"]["nickname"],
+                "name" : opponent["team"]["displayName"]
+            },
+            "home_away" : homeaway,
+            "location" : {
+                "full_name" : next_game["competitions"][0]["venue"]["fullName"],
+                "address" : next_game["competitions"][0]["venue"]["address"]
+            },
+            "status" : status
+        }
+    except IndexError:
```

Comment:
Is offseason the only time an `IndexError` can occur?

Reply:
It would be better if there was a more robust way to check if it's offseason.

Reply:
Yeah... the IndexError can only occur when there isn't a next game, therefore implying it's the offseason. I'll add a comment specifying this.

I ran a couple of sample requests on the ESPN API with teams whose last game had already been played (most notably Pitt basketball), and the nextEvent field in the response was their last game of the previous season. In that case, our API will return a response with the game info but status "GAME_COMPLETE," which seems to be more common for the offseason. 

All this to say the IndexError simply covers an edge case that could potentially occur.